os/tools/build_test.sh

```diff
@@ -45,26 +55,48 @@ for target in ${build_targets[@]}; do
 
 	# configure
 	echo "===================================="
-	echo "====== Configure \"${target}\" ======"
+	echo "====== Configure \"${build_targets[$index]}\" ======"
 	echo "===================================="	
-	cd ${TOOLDIR}; ./configure.sh ${target}
+	cd ${TOOLDIR}; ./configure.sh ${build_targets[$index]}
 
 	# build
 	echo "===================================="
-	echo "======== Build \"${target}\" ========"
-	echo "===================================="	
-	cd ${OSDIR}; ${BUILD_CMD} || { echo "\"${target}\" ERROR!!!"; exit 1; }
-
-	echo "===================================="
-	echo "======== \"${target}\" Built ========"
+	echo "======== Build \"${build_targets[$index]}\" ========"
 	echo "===================================="	
+	cd ${OSDIR};
+	OUTPUT_PATH=${OUTPUT_DIR}"`echo ${build_targets[$index]} | sed -e "s/\//\./g"`"
+	set -o pipefail;
+	${BUILD_CMD} |& tee ${OUTPUT_PATH}
+	if [ $? -eq 0 ]; then
+		echo "===================================="
+		echo "======== \"${build_targets[$index]}\" Built ========"
+		echo "===================================="
+		build_results[${index}]="SUCCESS"
+	else
+		is_failed=true
+		build_results[${index}]="FAIL"
+		echo "===================================="
+		echo "======== \"${build_targets[$index]}\" ERROR!!! ========"
+		echo "===================================="
+	fi
 done
 
 echo "===================================="
 echo "======== Clean build output ========"
 echo "===================================="
 cd ${OSDIR}; ${CLEAN_CMD}
 
-echo "===================================="
-echo "====== Build Test All SUCCESS ======"
-echo "===================================="
+echo "=========================================="
+echo "=========== Build Test Result ==========="
+echo "=========================================="
+for index in ${!build_targets[@]}; do
+	printf "%30s ------ %s\n" ${build_targets[$index]} ${build_results[$index]}
```

Comment:
Most of the prints are from `echo`, but you used `printf` in here only.
Is there any special reason?

Reply:
I used printf because of alignment of result logs for readability.